src/middlewares/logger.middleware.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable, Logger, NestMiddleware } from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class LoggerMiddleware implements NestMiddleware {
+  private readonly logger = new Logger();
+  private readonly jwt = new JwtService();
+
+  use(req: Request, res: Response, next: NextFunction) {
+    const token = req.headers?.authorization?.split(' ')[1] || '';
+    const userDetail = this.jwt.decode(token) || {};
```

Comment:
PullProofAI

There is a potential issue with directly accessing array elements without proper validation in the `LoggerMiddleware`. It's recommended to add checks to ensure that the authorization header and its split result are valid before accessing index 1.

```typescript
const token = (req.headers?.authorization || '').split(' ')[1] || '';
```



<!-- This is an auto-generated comment by PullProofAI -->